### DISCUSSION TITLE
Does AutoMapper work with Init Only Setters

### DISCUSSION THREAD

**Discussion by mpodonyi** [LuckyPennySoftware/AutoMapper#4064] [NA]
Can I map Models which have Init Only Setters and what I need to be aware of?

I am planning to refactor the properties of my models To "Init Only".
I don't want to get any runtime surprises.

Will I get compile time Errors when AutoMapper can not deal with "init only" setters?

Will "AssertConfigurationIsValid" inform me when Init Only Setter will fail?

**Comment by jbogard** [comment]
Give it a try and see. AutoMapper supports record types just fine, so I don't see why init-only properties would present a problem.
